Src/LogData.c

```diff
@@ -36,316 +73,209 @@ void writeToEEPROM(uint8_t* buffer, uint16_t bufferSize, uint16_t memAddress, ui
  * @brief Reads data from the EEPROM over I2C.
  * @param receiveBuffer, pointer to the buffer to store received data.
  * @param bufferSize, size of data to be read.
- * @param timeout, time to wait for read operation before timing out in ms.
  */
-void readFromEEPROM(uint8_t* receiveBuffer, uint16_t bufferSize, uint16_t memAddress, uint16_t timeout)
+void readFromEEPROM(uint8_t* receiveBuffer, uint16_t bufferSize, uint16_t memAddress)
 {
 	HAL_I2C_Mem_Read(&hi2c1, deviceAddress<<1, memAddress, sizeof(memAddress), receiveBuffer, bufferSize, timeout);
 }
 
 /**
  * @brief Blocks the current thread until the I2C state is ready for reading/writing.
- * @param timeout Timeout duration
  */
-void checkEEPROMBlocking(uint16_t timeout)
+void checkEEPROMBlocking()
 {
 	while(HAL_I2C_GetState(&hi2c1) != HAL_I2C_STATE_READY){};
 	while (HAL_I2C_IsDeviceReady(&hi2c1, deviceAddress<<1, 3, timeout) != HAL_OK){};
```

Comment:
We should maybe add a small (~100us?) delay in these spinlocks. It will noticeably reduce power consumption iirc. Thoughts? Also interested if the compiler would optimize these out.